modules/protocol/src/sync.ts

```diff
@@ -44,77 +38,35 @@ export async function outbound(
-  const generatePromise = (updateToSend: ChannelUpdate<any>, prevUpdate: ChannelUpdate<any> | undefined) =>
-    new Promise<Result<FullChannelState, ChannelUpdateError>>((resolve) => {
-      // If there is an error event corresponding to this channel and
-      // this nonce, reject the promise
-      errorEvt
-        .pipe((e: ChannelUpdateError) => {
-          return e.update?.channelAddress === updateToSend.channelAddress;
-        })
-        .attachOnce((e: ChannelUpdateError) => resolve(Result.fail(e)));
-
-      // If there is a channel update event corresponding to
-      // this channel update, resolve the promise
-      stateEvt
-        .pipe((e: FullChannelState) => {
-          return e.channelAddress === updateToSend.channelAddress;
-        })
-        .attachOnce((state: FullChannelState) => resolve(Result.ok(state)));
-
-      messagingService
-        .send(updateToSend.toIdentifier, {
-          to: updateToSend.toIdentifier,
-          from: updateToSend.fromIdentifier,
-          data: { update: updateToSend, latestUpdate: prevUpdate },
-        })
-        .catch((e) =>
-          resolve(
-            Result.fail(
-              new ChannelUpdateError(ChannelUpdateError.reasons.MessageFailed, updateToSend, storedChannel, {
-                error: e.message,
-              }),
-            ),
-          ),
-        );
-    });
-
-  // Retry sending the message 5 times w/3s delay
-  const sendWithRetry = async (
-    updateToSend: ChannelUpdate<any>,
-    prevUpdate: ChannelUpdate<any> | undefined,
-  ): Promise<Result<FullChannelState<any>, ChannelUpdateError>> => {
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    for (const _ of Array(5).fill(0)) {
-      const result = await generatePromise(updateToSend, prevUpdate);
-      if (!result.isError) {
-        logger.info({ method: "sendWithRetry", step: "Got result", result: result.getValue() });
-        return result;
-      }
-      const channelError = result.getError()!;
-      // The only errors we should retry here is if the message failed
-      // to send
-      if (channelError.message !== ChannelUpdateError.reasons.MessageFailed) {
-        return Result.fail(new ChannelUpdateError(channelError.message, channelError.update, channelError.state));
-      }
-      logger.error(`Failed to execute helper`, { error: result.getError()?.message, stack: result.getError()?.stack });
-      await delay(3_000);
-    }
+  logger.info({ method: "outbound", to: update.toIdentifier, type: update.type }, "Sending protocol message");
+  const result = await messagingService.sendProtocolMessage(update, storedChannel?.latestUpdate);
 
-    return Result.fail(new ChannelUpdateError(ChannelUpdateError.reasons.MessageFailed, updateToSend, storedChannel));
-  };
+  if (result.isError) {
+    logger.error({ method: "outbound", error: result.getError()! }, "Error receiving response!");
+    return Result.fail(result.getError()!);
```

Comment:
can't you just `return result`

Reply:
No, because `result` has a different type of value from the called function.